test/extended/core.sh

```diff
@@ -12,44 +12,87 @@ OS_ROOT=$(dirname "${BASH_SOURCE}")/../..
 source "${OS_ROOT}/hack/util.sh"
 source "${OS_ROOT}/hack/common.sh"
 os::log::install_errexit
+cd "${OS_ROOT}"
 
 ensure_ginkgo_or_die
 ensure_iptables_or_die
 
 os::build::setup_env
-go test -c ./test/extended -o ${OS_OUTPUT_BINPATH}/extended.test
+if [[ -z ${TEST_ONLY+x} ]]; then
+  go test -c ./test/extended -o ${OS_OUTPUT_BINPATH}/extended.test
+fi
 
-function cleanup()
-{
-  out=$?
-  cleanup_openshift
-  echo "[INFO] Exiting"
-  exit $out
-}
+export TMPDIR="${TMPDIR:-"/tmp"}"
+export BASETMPDIR="${TMPDIR}/openshift-extended-tests/core"
+export EXTENDED_TEST_PATH="${OS_ROOT}/test/extended"
+export KUBE_REPO_ROOT="${OS_ROOT}/../../../k8s.io/kubernetes"
 
-echo "[INFO] Starting 'images' extended tests"
+function join { local IFS="$1"; shift; echo "$*"; }
 
-trap "exit" INT TERM
-trap "cleanup" EXIT
+# The following skip rules excludes upstream e2e tests that fail.
+SKIP_TESTS=(
```

Comment:
There isn't when we run the unified binary.  The problem seems to be that
we are trying to use ginkgo in a way it doesn't work with.  Importing
upstream e2e for utility methods is problematic.  It also looks like we're
importing far too much (the final extended binary is 80M vs 95M for the
Openshift binary).  We should switch to explicit package execution, but in
order to do that we have to stop importing e2e and doing static init of the
suite.  Then we can run Kube separately.  We can switch to package
execution now probably, but I was still seeing  e2e run.

On Sep 21, 2015, at 9:44 AM, David Eads <notifications@github.com> wrote:

In test/extended/core.sh
<https://github.com/openshift/origin/pull/4732#discussion_r39971083>:

>
> -trap "exit" INT TERM
> -trap "cleanup" EXIT
> +# The following skip rules excludes upstream e2e tests that fail.
> +SKIP_TESTS=(

What's the rule for adding to this list during rebases? Since its raw name
matching, I worry about false matches here. Is there any way to run the
ginkgo tests via packages? That could ensure that everything in
openshift/origin was run.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4732/files#r39971083>.